erpnext/stock/report/stock_ledger/stock_ledger.py

```diff
@@ -55,6 +60,26 @@ def execute(filters=None):
 	update_included_uom_in_report(columns, data, include_uom, conversion_factors)
 	return columns, data
 
+def update_available_serial_nos(available_serial_nos, sle):
+	serial_nos = get_serial_nos(sle.serial_no)
+	key = (sle.item_code, sle.warehouse)
+	if key not in available_serial_nos:
+		available_serial_nos.setdefault(key, [])
+
+	existing_serial_no = available_serial_nos[key]
+	for sn in serial_nos:
+		if sle.actual_qty > 0:
+			if sn in existing_serial_no:
```

Comment:
If exists why should I remove it?

Reply:
I have seen some cases where delivery was made first and then receive of the serial nos, to handle that case I wrote this code.